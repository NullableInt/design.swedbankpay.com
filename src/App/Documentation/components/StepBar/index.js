import React from "react";
import PrismCode from "react-prism";

import { ComponentPreview, DocToc } from "#";
import StepBarComponent from "@/StepBar";

const BasicStepBar = () => {
    const items = [
        {
            title: "Step one"

        },
        {
            title: "Step two",
            active: true
        },
        {
            title: "Step three"
        },
        {
            title: "Step four"
        }
    ];

    return (
    <>
        <h2 id="basic-usage">Basic usage</h2>
        <p>Add class <code className="token property">.step-bar</code> to a containing <PrismCode className="language-html">{"<ol>"}</PrismCode> element and nest a <PrismCode className="language-html">{"<a>"}</PrismCode> or <PrismCode className="language-html">{"<span>"}</PrismCode> element inside the <PrismCode className="language-html">{"<li>"}</PrismCode> elements.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <StepBarComponent items={items} />
        </ComponentPreview>
    </>
    );
};

const StepBarText = () => (
    <div className="col-12 col-lg-10 doc-body">
        <p className="lead">Step Bar.</p>
        <BasicStepBar />
    </div>
);

const StepBar = () => (
    <div className="doc-container">
        <div className="row">
            <StepBar />
            <DocToc component={StepBarText} />
        </div>
    </div>
);

export default StepBar;

/* for testing */
export { BasicStepBar, StepBarText };
