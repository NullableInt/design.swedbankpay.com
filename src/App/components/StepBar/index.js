import React from "react";
import PropTypes from "prop-types";

const _renderItems = items => items.map(({ title, active }, i) => (
    <li key={i} className={active ? "active" : null}>
        {title}
        test
    </li>
));

const DefaultItems = () => (
    <div>
        <li>
            <div className="material-icons step-bar-icon" >check</div>
            Some title
            <div className="step-bar-item-sub-title">Some subtitle</div>
        </li>
        <li>
            Some title
            <div className="step-bar-item-sub-title">Some subtitle</div>
        </li>
        <li>
            Some title
        </li>
        <li>
            Some title
        </li>
    </div>
);

const StepBar = ({ items, horizontal }) => (
    <ol className={`step-bar${horizontal ? " step-bar-horizontal" : ""}`}>
        {items ? _renderItems(items) : <DefaultItems />}
    </ol>
);

StepBar.propTypes = {
    items: PropTypes.array,
    horizontal: PropTypes.bool
};

export default StepBar;

// exporting for testing purposes
export { DefaultItems };
