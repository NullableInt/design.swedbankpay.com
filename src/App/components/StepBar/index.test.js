import React from "react";
import { shallow } from "enzyme";

import StepBar, { DefaultItems } from "./index";

describe("Component: StepBar", () => {
    it("is defined", () => {
        expect(StepBar).toBeDefined();
    });

    it("renders with default items when no items is provided through items prop", () => {
        const wrapper = shallow(
            <StepBar />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.html()).toContain("step-bar");
        expect(wrapper.contains(<DefaultItems />)).toEqual(true);
    });
});
