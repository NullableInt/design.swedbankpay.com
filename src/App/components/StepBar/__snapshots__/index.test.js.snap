// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: StepBar renders with default items when no items is provided through items prop 1`] = `
<ol
  className="step-bar"
>
  <DefaultItems />
</ol>
`;
